sdk/nodejs/provider/experimental/analyzer.ts

```diff
@@ -584,6 +588,26 @@ function isInput(type: typescript.Type): boolean {
     return hasOutput && hasPromise && hasOther;
 }
 
+function isResourceReference(type: typescript.Type, checker: typescript.TypeChecker): boolean {
+    if (type.isClassOrInterface()) {
```

Comment:
Nit, but it's better to short-circuit and return if it's not isClassOrInterface (why interface by the way?)

Reply:
Yes, and I think it can only be `isClass` as well. Changed both things.